common/src/main/java/io/pravega/example/tensorflow/FaceRecognizer.java

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright (c) 2019 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ */
+package io.pravega.example.tensorflow;
+
+
+import io.pravega.example.common.Embedding;
+import io.pravega.example.common.VideoFrame;
+import org.apache.commons.io.IOUtils;
+import org.bytedeco.opencv.opencv_core.*;
+import org.bytedeco.opencv.opencv_objdetect.CascadeClassifier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.tensorflow.*;
+
+import java.io.*;
+import java.net.URL;
+import java.nio.FloatBuffer;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static org.bytedeco.opencv.global.opencv_core.*;
+import static org.bytedeco.opencv.global.opencv_imgcodecs.*;
+import static org.bytedeco.opencv.global.opencv_imgproc.*;
+import static org.bytedeco.opencv.global.opencv_objdetect.CASCADE_SCALE_IMAGE;
+
+/**
+ * ObjectDetector class to detect and recognize faces using pre-trained models with TensorFlow Java API.
+ */
+public class FaceRecognizer implements Serializable, Closeable {
```

Comment:
This class does two very different things. It calculates embeddings using TensorFlow and it finds matching embeddings. You should split these operations into different classes.

Reply:
I dont get this. Matching process is like a compareTo right? and its not a lot of code to be separated right?

Reply:
So, I incorporated this change by separating the matching process into separate class. I am wondering if I should not use FaceRecognizer class in FaceRecognizerProcessor class because FaceRecognizer uses tensorflow. I could use a map function to process the transactions into embeddings if you think its a good idea.